src/utils/transaction.ts

```diff
@@ -134,13 +135,25 @@ const transaction = {
     return signature.toDER();
   },
 
-  async getSignatureForTx(tx: Transaction, storage: IStorage, pinCode: string): Promise<IInputSignature[]> {
+  /**
+   * Get the signatures for a transaction
+   * @param tx Transaction to sign
+   * @param storage Storage of the wallet
+   * @param pinCode Pin to unlock the mainKey for signatures
+   */
+  async getSignatureForTx(tx: Transaction, storage: IStorage, pinCode: string): Promise<ITxSignatureData> {
     const xprivstr = await storage.getMainXPrivKey(pinCode);
     const xprivkey = HDPrivateKey.fromString(xprivstr);
     const dataToSignHash = tx.getDataToSignHash();
     const signatures: IInputSignature[] = [];
+    let ncCallerSignature: Buffer|null = null;
 
     for await (const {tx: spentTx, input, index: inputIndex} of storage.getSpentTxs(tx.inputs)) {
+      if (input.data) {
```

Comment:
codecov is complaining this is not covered by tests, should it be? I feel it is a fundamental part of this method and should fail if this behavior is changed in the future

Reply:
Created a test for this